src/database/sqlite3/sqlite_database.cc

```diff
@@ -246,19 +246,16 @@ std::shared_ptr<SQLResult> Sqlite3Database::select(const std::string& query)
         return stask->getResult();
     } catch (const std::runtime_error& e) {
         if (!shutdownFlag) {
-            if (dbInitDone) {
-                log_error("Prematurely shutting down.\n{}", e.what());
-                rollback("");
-                shutdown();
-            }
-            throw_std_runtime_error(e.what());
-        } else {
-            if (dbInitDone) {
-                log_error("Already shutting down.\n{}", e.what());
-                return {};
-            } else
+            if (!dbInitDone)
```

Comment:
Maybe this can be simpler if you check `dbInitDone` first

Reply:
I think this is what you meant?